codex-ui/src/vue/components/alert/BaseAlert.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <div
+    ref="el"
+    @click="clickHandler"
+  >
+    <div
+      class="notex-alert"
+      :class="classes"
+    >
+      <Icon
+        v-if="icon"
+        :name="icon"
+      />
+      <div>{{ props.content }}</div>
+      <span
+        :class="`${ALERT_NAMESPACE}__close-button`"
+        @click="closeAlert"
+      >&times;
+      </span>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { defineProps, computed, withDefaults, ref, watch, nextTick } from 'vue';
+import { useElementHover, useFocus, useDraggable } from '@vueuse/core';
+import Icon from '../icon/Icon.vue';
+import { AlertOptions } from './core/types';
+import { ALERT_CONTAINER_DEFAULTS, ALERT_NAMESPACE } from './core/constant';
+import { useAlert } from './core/useAlert';
+
+const el = ref<HTMLElement>();
+
+const props = withDefaults(defineProps<AlertOptions>(), {
+  id: ALERT_CONTAINER_DEFAULTS.id,
+  position: ALERT_CONTAINER_DEFAULTS.position,
+  content: ALERT_CONTAINER_DEFAULTS.content,
+  closeOnClick: ALERT_CONTAINER_DEFAULTS.closeOnClick,
+  icon: ALERT_CONTAINER_DEFAULTS.icon,
+  status: ALERT_CONTAINER_DEFAULTS.status,
+  timeout: ALERT_CONTAINER_DEFAULTS.timeout,
+  onClick: ALERT_CONTAINER_DEFAULTS.onClick,
+  onClose: ALERT_CONTAINER_DEFAULTS.onClose,
+});
+
+const { value: hovering } = useElementHover(el);
+const { focused } = useFocus(el);
+const { isDragging } = useDraggable(el);
+
+/**
+ * a progress bar can be provided using this computed property
+ */
+// eslint-disable-next-line no-unused-vars
+const runingAlert = computed(() => !hovering && focused.value && !isDragging.value);
+const { alertStore } = useAlert(props);
+
+/**
+ * close alert
+ */
+const closeAlert = () => {
+  for (var i = alertStore.value.length - 1; i >= 0; i--) {
+    if (alertStore.value[i].id === props.id) {
+      alertStore.value.splice(i, 1);
+    }
+  }
+};
+
+watch(isDragging, (v) => {
+  if (v !== null) {
+    nextTick(() => closeAlert());
+  }
+});
+
+/**
+ * handles closure and dragging of alert box
+ */
+const clickHandler = () => {
+  if (props.onClick) {
+    props.onClick(closeAlert);
+  }
+
+  if (props.closeOnClick) {
+    closeAlert();
+  }
+};
+
+const classes = computed(() => {
+  return [`${ALERT_NAMESPACE}`, `${ALERT_NAMESPACE}--${props.status}`, `${props.position}`];
+});
+</script>
+
+<style  lang="postcss">
+.notex-alert {
+  --padding-left: var(--h-padding);
+  --padding-right: var(--h-padding);
+  --color: var(--accent--text-solid-foreground);
+  --success: #00A64D;
+  --error: #BB393D;
+  --warning: #DE6205;
+  --info: #54617B;
+  --default: #54617B;
```

Comment:
colors should not be hardcoded, use solid color and direct theme specifying through the "theme-base" attribute. We have a themes for all colors in figma: "red", "amber", "grass", "graphite" (for info) and the default alert should use a base theme